src/shared/logger/activation.ts

```diff
@@ -19,113 +19,54 @@ import { WinstonToolkitLogger } from './winstonToolkitLogger'
-        })
-    } else {
-        outputChannel = params.outputChannel
-        logPath = params.logPath
+    await ensureLogFolderExists(path.dirname(logPath))
 
-        setLogger(createLogger(params))
-    }
+    setLogger(makeLogger(logLevel, logPath, outputChannel))
 
-    if (outputChannel && logPath) {
-        outputChannel.appendLine(
-            localize('AWS.log.fileLocation', 'Error logs for this session are permanently stored in {0}', logPath)
-        )
-    }
+    await registerLoggerCommands()
 
-    return getLogger()
+    outputChannel.appendLine(
+        localize('AWS.log.fileLocation', 'Error logs for this session are permanently stored in {0}', logPath)
+    )
 }
 
-/**
- * @param params: LoggerParams--nothing is required, but a LogPath is highly recommended so Winston doesn't throw errors
- *
- * Outputs a logger object that isn't stored anywhere--it's up to the caller to keep track of this.
- */
-export function createLogger(params: LoggerParams): Logger {
-    // TODO : log level should be a concern of the caller
-    let level: LogLevel
-    if (params.logLevel) {
-        level = params.logLevel
-    } else {
-        const configuration: SettingsConfiguration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
-        const setLevel = configuration.readSetting<string>('logLevel')
-        level = setLevel ? (setLevel as LogLevel) : DEFAULT_LOG_LEVEL
-    }
+export function makeLogger(logLevel: LogLevel, logPath: string, outputChannel: vscode.OutputChannel): Logger {
+    const logger = new WinstonToolkitLogger(logLevel)
+    logger.logToFile(logPath)
+    logger.logToOutputChannel(outputChannel)
 
-    const logger = new WinstonToolkitLogger(level)
-    if (params.logPath) {
-        logger.logToFile(params.logPath)
-    }
+    return logger
+}
 
-    if (params.outputChannel) {
-        logger.logToOutputChannel(params.outputChannel)
-    }
+function getLogLevel(): LogLevel {
+    const configuration: SettingsConfiguration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
 
-    return logger
+    return configuration.readSetting<LogLevel>('logLevel', DEFAULT_LOG_LEVEL)
 }
 
-function getDefaultLogPath(): string {
+function getLogBasePath(): string {
     if (os.platform() === 'win32') {
-        return path.join(os.homedir(), 'AppData', 'Roaming', LOG_RELATIVE_PATH, DEFAULT_LOG_NAME)
+        return path.join(os.homedir(), 'AppData', 'Roaming')
     } else if (os.platform() === 'darwin') {
-        return path.join(os.homedir(), 'Library', 'Application Support', LOG_RELATIVE_PATH, DEFAULT_LOG_NAME)
+        return path.join(os.homedir(), 'Library', 'Application Support')
     } else {
-        return path.join(os.homedir(), '.config', LOG_RELATIVE_PATH, DEFAULT_LOG_NAME)
+        return path.join(os.homedir(), '.config')
     }
 }
 
-function makeDefaultLogName(): string {
+function makeLogFilename(): string {
```

Comment:
I would recommend changing this to:
```
function makeLogFilename(filename: string = 'aws_toolkit_'): string {
```
You wouldn't want to name non-default logs with this syntax, unless this is only for the default logger?

Reply:
There is only one toolkit log, with only one filename convention. Supporting alternate prefixes is not necessary.